libursa/src/ffi/cl/verifier.rs

```diff
@@ -144,7 +183,7 @@ pub extern "C" fn ursa_cl_proof_verifier_verify(
     check_useful_c_reference!(nonce, Nonce, ErrorCode::CommonInvalidParam3);
     check_useful_c_ptr!(valid_p, ErrorCode::CommonInvalidParam4);
 
-    let proof_verifier = unsafe { Box::from_raw(proof_verifier as *mut ProofVerifier) };
+    let mut proof_verifier = unsafe { Box::from_raw(proof_verifier as *mut ProofVerifier) };
```

Comment:
This should be:

```
let mut proof_verifier = serde_json::deserialize::<ProofVerifier>(buf);
```
using unsafe is...uh...*unsafe*.

Reply:
I am following the pattern being followed in the FFI code in this project.